src/api/menu.ts

```diff
@@ -41,6 +43,12 @@ export function validateMenu(menu: Menu) {
     errors.push("menu title must be a string");
   }
 
+  if (menu.menu_type !== undefined && !(menu.menu_type == 'hotkey' || menu.menu_type == 'palette')) {
```

Comment:
nit (here and below): please use `===`
```suggestion
  if (menu.menu_type !== undefined && !(menu.menu_type === 'hotkey' || menu.menu_type === 'palette')) {
```

nit: IMO not having a negated boolean expression is easier to read
```suggestion
  if (menu.menu_type !== undefined && menu.menu_type !== 'hotkey' && menu.menu_type !== 'palette') {
```